amd-power-control: Turin Lenovo set FP power LED

modolaty opened the pull request:
Set Front Panel Power LED for Volcano and Purico Platforms based on Power State On or Off

modolaty committed:
amd-power-control: Turin Lenovo set FP power LED

Set Front Panel Power LED for Volcano and Purico Platforms
based on Power State On or Off

Signed-off-by: Mohsen Dolaty <mohsen.dolaty@amd.com>

supven01 commented on power-control-x86/src/power_control.cpp:
```diff
@@ -144,18 +163,57 @@ static int getGPIOValue(const std::string& name)
     switch (state)
     {
         case PowerState::on:
+            setFrontpanelPowerLed(true);
```
Check power on/off multiple times as the app is time sensitive

modolaty replied:
done. in both Purico and Volcano

supven01 commented on power-control-x86/src/power_control.cpp:
```diff
@@ -144,18 +163,57 @@ static int getGPIOValue(const std::string& name)
+    char i2c_devname[FILEPATHSIZE];
+    int portRead, portWrite, dataRead, dataWrite;
+
+    // check volcano FP
```
This logic needs to be broken down.

//create a new function i2ccommandLed(char *i2c_devname) or similar

setFrontpanelPowerLed()
{
   // Attempt bus 1
i2ccommandLed(bus1);
   // Attempt bus 2
i2ccommandLed(bus2);
}

Ideally, we would implement as virtual void setFrontpanelPowerLed();
and override setFrontpanelPowerLed in individual platform files, but this cannot be done quick. Hence the above approach.

modolaty replied:
done

modolaty committed:
Update power_control.cpp

created new function i2cCommandLed() and separate Volcano from Purico so we can replace the call in meta-turin later on

supven01 commented on power-control-x86/src/power_control.cpp:
```diff
@@ -192,11 +186,21 @@ static void setFrontpanelPowerLed(bool led)
+static void setFrontpanelPowerLed(bool led)
+{
+    // check Volcano FP
+    if (i2cCommandLed(VOLCANO_FP_I2C_BUS, led) == false) {
```
Lets not create dependency between Volcano and Purico.
if (i2cCommandLed(VOLCANO_FP_I2C_BUS, led) == false)
{
   //print message
}

if (i2cCommandLed(PURICO_FP_I2C_BUS, led) == false)
{
  // Print message
 }

modolaty replied:
done, did not print messages to avoid confusion

modolaty committed:
Update power_control.cpp

separate Volcano from Purico
